tests/experimental/metric_repository/test_column_descriptive_metrics_metric_retriever_integration.py

```diff
@@ -64,88 +75,127 @@ def test_get_metrics(
-            column="col1",
+            column="numeric_with_nulls_1",
             value=2,
             exception=None,
         ),
         ColumnMetric[float](
             batch_id=batch_id,
             metric_name="column.max",
-            column="col2",
+            column="numeric_with_nulls_2",
             value=4,
             exception=None,
         ),
         ColumnMetric[float](
             batch_id=batch_id,
             metric_name="column.mean",
-            column="col1",
+            column="numeric_with_nulls_1",
             value=1.5,
             exception=None,
         ),
         ColumnMetric[float](
             batch_id=batch_id,
             metric_name="column.mean",
-            column="col2",
+            column="numeric_with_nulls_2",
             value=3.5,
             exception=None,
         ),
         ColumnMetric[float](
             batch_id=batch_id,
             metric_name="column.median",
-            column="col1",
+            column="numeric_with_nulls_1",
             value=1.5,
             exception=None,
         ),
         ColumnMetric[float](
             batch_id=batch_id,
             metric_name="column.median",
-            column="col2",
+            column="numeric_with_nulls_2",
             value=3.5,
             exception=None,
         ),
         TableMetric[List[str]](
             batch_id=batch_id,
             metric_name="table.column_types",
             value=[
-                {"name": "col1", "type": "float64"},
-                {"name": "col2", "type": "float64"},
+                {"name": "numeric_with_nulls_1", "type": "float64"},
+                {"name": "numeric_with_nulls_2", "type": "float64"},
+                {"name": "string", "type": "object"},
+                {"name": "string_with_nulls", "type": "object"},
+                {"name": "boolean", "type": "bool"},
+                {"name": "datetime", "type": "datetime64[ns]"},
             ],
             exception=None,
         ),
         ColumnMetric[int](
             batch_id=batch_id,
             metric_name="column_values.null.count",
-            column="col1",
+            column="numeric_with_nulls_1",
             value=1,
             exception=None,
         ),
         ColumnMetric[int](
             batch_id=batch_id,
             metric_name="column_values.null.count",
-            column="col2",
+            column="numeric_with_nulls_2",
             value=1,
             exception=None,
         ),
+        ColumnMetric[int](
+            batch_id=batch_id,
+            metric_name="column_values.null.count",
+            column="string",
+            value=0,
+            exception=None,
+        ),
+        ColumnMetric[int](
+            batch_id=batch_id,
+            metric_name="column_values.null.count",
+            column="string_with_nulls",
+            value=1,
+            exception=None,
+        ),
+        ColumnMetric[int](
+            batch_id=batch_id,
```

Comment:
ensure `boolean` column and `datetime` column do not contain numeric metrics like `mean` etc.